lib/hooks/LocalStorage/index.js

```diff
@@ -0,0 +1,62 @@
+import React from 'react'
+
+/**
+Hook to store a persistent state in local storage.
+
+The state is shared among any component using it
+from any tab of the same browser
+accessing any application hosted in the same domain.
+
+A null value means the value is unset.
+Setting the state to null unsets it.
+Any other value is converted to string including undefined.
+The initial value is set only if the value is unset
+on component initialization.
+Initial value of undefined is ignored.
+*/
+
+export default function useLocalStorage(key, initialValue) {
+  function decode(json) {
+    const value = json === null ? null : JSON.parse(json)
+    return value
+  }
+
+  function encodeValue(value) {
+    const json = value === null ? null : JSON.stringify(value)
+    return json
+  }
+
+  function setStorage(value) {
+    if (value === null) localStorage.removeItem(key)
+    else localStorage.setItem(key, encodeValue(value))
+  }
+
+  const [state, setState] = React.useState(() => {
+    const storedValue = localStorage.getItem(key)
+    if (storedValue === null) {
+      if (initialValue !== undefined) {
+        setStorage(initialValue)
+      }
+    }
+    return decode(localStorage.getItem(key))
+  })
+  // Update state when changed from other tab
+  React.useEffect(() => {
+    const listener = (e) => {
+      if (e.key === key) {
+        setState(decode(e.newValue))
+      }
+    }
+    window.addEventListener('storage', listener)
+    return () => {
+      window.removeEventListener('storage', listener)
+    }
+  }, [])
```

Comment:
Do we have to add `key` as a dependency to ensure reactivity when the key changes?

Reply:
I don't undestand the problem, could you develop?

Reply:
The way it is used, I don't see it as necessary.